File: src/main/java/net/dankrushen/danknn/danklayers/DankInputLayer.java
package net.dankrushen.danknn.danklayers;

import net.dankrushen.danknn.DankConnection;

public class DankInputLayer extends DankLayer implements IDankInputLayer {
    private IDankOutputLayer outputLayer;

    public DankInputLayer(int numNeurons) {
        super(numNeurons);
    }

    @Override
    public void setOutputLayer(IDankOutputLayer outputLayer) {
        this.outputLayer = outputLayer;
    }

    @Override
    public IDankOutputLayer getOutputLayer() {
        return outputLayer;
    }

    @Override
    public DankConnection[] getOutputConnections() {
        return outputLayer.getInputConnections();
    }
}

File: src/main/java/net/dankrushen/danknn/danklayers/DankInputLayer.kt
package net.dankrushen.danknn.danklayers

import net.dankrushen.danknn.DankConnection

class DankInputLayer(numNeurons: Int) : DankLayer(numNeurons), IDankInputLayer, Cloneable {
    override lateinit var outputLayer: IDankOutputLayer

    override val outputConnections: Array<DankConnection>
        get() = outputLayer.inputConnections

    public override fun clone(): DankInputLayer {
        val clone = DankInputLayer(neurons.size)

        for (i in neurons.indices) {
            neurons[i].copyTo(clone.neurons[i])
        }

        return clone
    }
}
